Nodejs/Product/Nodejs/Debugger/NodeBreakpoint.cs

```diff
@@ -55,18 +59,53 @@ sealed class NodeBreakpoint {
         /// This translates the breakpoint from the location where the user set it (possibly
         /// a TypeScript file) into the location where it lives in JavaScript code.
         /// </summary>
-        public FilePosition GetPosition(SourceMapper mapper) {
+        public FilePosition GetPosition(string javaScriptFileName) {
             // Checks whether source map is available
-            string javaScriptFileName;
+
             int javaScriptLine;
             int javaScriptColumn;
+            DecodedSourceMap decodedSourceMap = null;
+            DkmTextSpan? jsSpan = null;
+            string sourceMapFilename;
+
+            sourceMapFilename = FindSourceMapFile(javaScriptFileName);
+            if (!string.IsNullOrEmpty(sourceMapFilename)) {
+                SourceMapReader sourceMapReader = new SourceMapReader();
+                decodedSourceMap = sourceMapReader.LoadSourceMap(javaScriptFileName, sourceMapFilename);
+
+                // The NodeJS debugger is 1-based index while the SourceMapReader is 0-based index
+                jsSpan = decodedSourceMap.MapTsSourcePosition(Target.FileName, new DkmTextSpan(Target.Line + 1, Target.Line + 1, Target.Column + 1, Target.Column + 1));
+                javaScriptLine = jsSpan.Value.StartLine - 1;
+                javaScriptColumn = jsSpan.Value.StartColumn - 1;
+                if (javaScriptLine >= 0 && javaScriptColumn >= 0) {
+                    return new FilePosition(javaScriptFileName,  javaScriptLine, javaScriptColumn);
+                }
+            }           
+
+             return Target;
+        }
 
-            if (mapper != null &&
-                mapper.MapToJavaScript(Target.FileName, Target.Line, Target.Column, out javaScriptFileName, out javaScriptLine, out javaScriptColumn)) {
-                return new FilePosition(javaScriptFileName, javaScriptLine, javaScriptColumn);
-            }
-
-            return Target;
+        string FindSourceMapFile(string jsFileName) {
+            string sourceMapFilename = null;
+
+            if (File.Exists(jsFileName)) {
+                string[] contents = File.ReadAllLines(jsFileName);
+                const string marker = "# sourceMappingURL=";
+                int markerStart;
+                string markerLine = contents.Reverse().FirstOrDefault(x => x.IndexOf(marker, StringComparison.Ordinal) != -1);

```

Comment:
This the the same logic used by the main SourceMap Reader. @sheetalkamat could shed more light on this.